lib/fumanchu/compiler.ex

```diff
@@ -3,11 +3,49 @@ defmodule FuManchu.Compiler do
   alias FuManchu.Parser
   alias FuManchu.Generator
 
+  @spec compile(String.t) :: (map -> String.t)
```

Comment:
Defining a type for a compiled template would make this spec a bit easier to understand, imo.

```elixir
@type compiled_template :: (%{} -> String.t)

@spec compile(String.t) :: compiled_template()
```